lisa/wa.py

```diff
@@ -598,16 +609,28 @@ def trace_idle_analysis(trace):
     NAME = 'trace'
     _ARTIFACT_NAME = 'trace-cmd-bin'
     _PURE_GET_JOB_DF = False
+    _FORCE_AUTO = True
 
-    def __init__(self, wa_output, trace_to_df, **kwargs):
+    def __init__(self, wa_output, trace_to_df=None, **kwargs):
         self._trace_to_df = trace_to_df
         self._trace_kwargs = kwargs
         super().__init__(wa_output, df_postprocess=None)
 
     def _get_artifact_df(self, path):
+        if self._trace_to_df is None:
+            raise ValueError('trace_to_df was not specified for the TraceCollector')
         trace = Trace(path, **self._trace_kwargs)
         return self._trace_to_df(trace)
 
+    @property
+    @memoized
```

Comment:
`memoized` is a bit dangerous as Traces will by default take arbitrary amounts of memory. That's a great way to end up with some sort of leak.

Reply:
Hmm should I just drop it then? Without it the traces take a good moment to load every time so it definitely makes the experience worse :/